examples/with-stomp/useClient.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { Stomp } from '@stomp/stompjs'
+
+let stompClient
+
+const useClient = () => {
+  const [client, setClient] = React.useState(stompClient)
+
+  React.useEffect(() => {
+    if (!stompClient) {
```

Comment:
Use a `useRef` and your hook is completly encapsulated.

Reply:
@StarpTech Hey Dustin, Can you provide a suggested change? I appreciate that.

Reply:
This is better. The unsubscribe code is pseudo code.
```js
const useClient = () => {
	// initiate client only on initial render
	const stompClient = React.useState(() => Stomp.client(process.env.STOMP_SERVER))
	useEffect(() => {
		//destroy client
		return () => stompClient.close()
	}, []); // unsubscribe on unmount
	return stompClient;
}
```

Reply:
Forget the useState usage it's wrong. Go with your previous solution it's good enough for an example.

Reply:
In this case let's not, `useRef` only persists the value until the component dies, but `let stompClient` will persist it until the page is refreshed, and that's the idea.

Reply:
@lfades you're right, that's wrong.